client/client-multi/client-cpp/kaa/KaaClientContext.hpp

```diff
@@ -26,27 +26,32 @@
 #include <kaa/IKaaClientStateStorage.hpp>
 #include <kaa/context/IExecutorContext.hpp>
 
-
 namespace kaa {
 
 class KaaClientContext : public IKaaClientContext
 {
 public:
-    KaaClientContext(KaaClientProperties &properties, ILogger &logger,
-                     IExecutorContext &executorContext, IKaaClientStateStoragePtr state = nullptr)
-	  : properties_(properties), logger_(logger), executorContext_(executorContext), state_(state) {}
-
-    virtual KaaClientProperties              &getProperties() { return properties_; }
-    virtual ILogger                              &getLogger() { return logger_; }
-    virtual IKaaClientStateStorage               &getStatus() { return *state_; }
-    virtual IExecutorContext            &getExecutorContext() { return executorContext_; }
+    KaaClientContext(KaaClientProperties &properties, ILogger &logger, IExecutorContext &executorContext,
+        IKaaClientStateListenerPtr stateListener, IKaaClientStateStoragePtr state = nullptr)
```

Comment:
Can we add default value for stateListener and move in the end, so we won't change all the tests?